src/comgr.cpp

```diff
@@ -0,0 +1,503 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/env.hpp>
+#include <miopen/errors.hpp>
+#include <miopen/logger.hpp>
+#include <miopen/stringutils.hpp>
+#include <amd_comgr.h>
+#include <algorithm>
+#include <exception>
+#include <cstddef>
+#include <cstring>
+#include <tuple> // std::ignore
+#include <vector>
+
+/// \todo see issue #1222, PR #1316
+MIOPEN_DECLARE_ENV_VAR(MIOPEN_DEBUG_SRAM_EDC_DISABLED)
+
+#define DEBUG_DETAILED_LOG 0
+#define COMPILER_LC 1
+
+#if DEBUG_DETAILED_LOG
+#define EC_BASE_DETAILED_INFO(comgrcall, info) \
+    MIOPEN_LOG_I("Ok \'" #comgrcall "\' " << to_string(info))
+#else
+#define EC_BASE_DETAILED_INFO(comgrcall, info)
+#endif
+
+#define EC_BASE(comgrcall, info, expr2)                                   \
+    do                                                                    \
+    {                                                                     \
+        const amd_comgr_status_t status = (comgrcall);                    \
+        if(status != AMD_COMGR_STATUS_SUCCESS)                            \
+        {                                                                 \
+            MIOPEN_LOG_E("\'" #comgrcall "\' " << to_string(info) << ": " \
+                                               << GetStatusText(status)); \
+            (expr2);                                                      \
+        }                                                                 \
+        else                                                              \
+            EC_BASE_DETAILED_INFO((comgrcall), (info));                   \
+    } while(false)
+
+#define EC(comgrcall) EC_BASE(comgrcall, NoInfo, (void)0)
+#define EC_THROW(comgrcall) EC_BASE(comgrcall, NoInfo, Throw(status))
+#define ECI_THROW(comgrcall, info) EC_BASE(comgrcall, info, Throw(status))
+#define ECI_THROW_MSG(comgrcall, info, msg) EC_BASE(comgrcall, info, Throw(status, (msg)))
```

Comment:
Can we make these function or put them inline instead of using macros? We dont do this for any other API calls.

Reply:
I believe we can, and do not like this ugly stuff too. Can't spend time for this refactoring right now, should I open a ticket?

Reply:
Will open ticket as soon as PR merged.